libs/designsystem/modal/src/modal/services/modal.helper.ts

```diff
@@ -134,25 +147,29 @@ export class ModalHelper {
 
     // Back button should only be handled manually
     // if the modal is not instantiated through a route.
-    if (!config.modalRoute && !config.canDismiss && !alertConfig) {
-      this.handleBrowserBackButton(ionModal);
+    if (!config.modalRoute) {
+      this.handleBrowserBackButton(overlay);
     }
 
     this.isModalOpening = false;
 
-    return {
-      dismiss: ionModal.dismiss.bind(ionModal),
-      onWillDismiss: ionModal.onWillDismiss(),
-      onDidDismiss: ionModal.onDidDismiss(),
-    };
+    return overlay;
   }
 
-  private handleBrowserBackButton(modal: HTMLIonModalElement) {
-    const popStateEvent$ = fromEvent(this.windowRef.nativeWindow, 'popstate').pipe(first());
-    const modalClose$ = fromEvent(modal, 'ionModalDidDismiss');
-
-    popStateEvent$.pipe(takeUntil(modalClose$)).subscribe(() => {
-      modal.dismiss();
-    });
+  private handleBrowserBackButton(overlay: Overlay) {
```

Comment:
We could consider extracting this function and share it with the component based modal...